pkg/storaged/lvol-tabs.jsx

```diff
@@ -49,57 +48,145 @@ function lvol_rename(lvol) {
     });
 }
 
-function lvol_resize(client, lvol) {
+function lvol_and_fsys_resize(client, lvol, size, offline) {
+    var block, crypto, fsys;
+    var crypto_overhead;
+    var vdo;
+
+    block = client.lvols_block[lvol.path];
+    if (!block)
+        return lvol.Resize(size, { });
+
+    crypto = client.blocks_crypto[block.path];
+    if (crypto) {
+        var cleartext = client.blocks_cleartext[block.path];
+        if (!cleartext)
+            return;
+        fsys = client.blocks_fsys[cleartext.path];
+        vdo = client.vdo_overlay.find_by_backing_block(cleartext);
+        crypto_overhead = block.Size - cleartext.Size;
+    } else {
+        fsys = client.blocks_fsys[block.path];
+        vdo = client.vdo_overlay.find_by_backing_block(block);
+        crypto_overhead = 0;
+    }
+
+    function fsys_resize() {
+        if (fsys) {
+            // When doing an offline resize, we need to first repair the filesystem.
+            if (offline) {
+                return fsys.Repair({ }).then(function () { return fsys.Resize(size - crypto_overhead, { }); });
```

Comment:
Does `Repair` unmount the file system if necessary? It's not [documented](http://storaged.org/doc/udisks2-api/2.7.4/gdbus-org.freedesktop.UDisks2.Filesystem.html#gdbus-method-org-freedesktop-UDisks2-Filesystem.Repair) to do that. There is no `Unmount()` call in this PR, but maybe it already happens in the existing code?

Reply:
The unmounting happens in `teardown_active_usage`.